src/fileioc/fileioc.asm

```diff
@@ -1573,6 +1626,29 @@ util_set_offset:
 	ld	(hl), bc
 	ret
 
+util_Arc_Unarc: ;properly handle garbage collects :P
+	call	_ChkFindSym
+	push	hl
+	call	_ChkInRAM
+	pop	hl
+	jr	nz,.arc_unarc ;if the file is already in archive, we won't trigger a gc
+	call	_LoadDEInd_s
+	ld	hl,12
+	add	hl,de
+	call	_FindFreeArcSpot ;check if we will trigger a gc
```

Comment:
There is already a routine that does this.

Reply:
optimized, now uses it